lib/domain/contact_manager/contacts_manager.dart

```diff
@@ -4,20 +4,39 @@ import 'package:fluffychat/config/app_config.dart';
 import 'package:fluffychat/di/global/get_it_initializer.dart';
 import 'package:fluffychat/domain/app_state/contact/get_contacts_state.dart';
 import 'package:fluffychat/domain/model/contact/contact.dart';
+import 'package:fluffychat/domain/model/contact/contacts_manager_agruments.dart';
 import 'package:fluffychat/domain/usecase/get_all_contacts_interactor.dart';
+import 'package:fluffychat/utils/permission_service.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
 import 'package:matrix/matrix.dart';
+import 'package:permission_handler/permission_handler.dart';
 
 enum SyncContactsState {
   initial,
   synchronizing,
   synchronized,
 }
 
+enum DisplayWaringContactsBannerState {
+  show,
+  hide,
+}
+
 class ContactsManager {
   SyncContactsState _syncContactsState = SyncContactsState.initial;
 
+  DisplayWaringContactsBannerState _displayWaringContactsBannerState =
+      PlatformInfos.isMobile
+          ? DisplayWaringContactsBannerState.show
+          : DisplayWaringContactsBannerState.hide;
+
+  bool _isNotDisplayWarningAgainCacheMemory = false;
+
   final _getAllContactsInteractor = getIt.get<GetAllContactsInteractor>();
 
+  final PermissionHandlerService permissionHandlerService =
+      PermissionHandlerService();
+
   Set<Contact> _tomContacts = {};
```

Comment:
why it is not `ValueNotifier`? It makes app freeze a lot. Update it whenever we complete it. Avoid 
![Uploading image.png…]()